src/main/scala/scalaz/netty/Client.scala

```diff
@@ -36,6 +36,7 @@ import _root_.io.netty.handler.codec._
 
 private[netty] final class Client(limit: Int) {
   // this isn't ugly or anything...
+  @volatile
```

Comment:
it can be managed by Guice, but what I meant is to essentially move both channel and queue to constructor. So we could move handler out of the server class scope, and channel in server does not need to be a var.